src/provider/application.js

```diff
@@ -334,6 +349,74 @@ class Application extends EventEmitter {
       this.trigger('xfc.unload');
     }
   }
+
+  /**
+   * When page load is completed, we need to check if the document has
+   * any interactable element or not. Additionally, we need to check
+   * if the iframe has scrolling enabled or not. This determines if we need
+   * set focus with `tabIndex = 0` in the document or not.
+   */
+  handleLoadEvent() {
+    this.hasInteractableElement = hasInteractableElement();
+
+    this.JSONRPC.request('isScrollingEnabled', [])
+      .then((scrolling) => { this.isScrollingEnabled = scrolling; });
+
+    if (this.isIframeScrollable() === true && isContentScrollable() && this.hasInteractableElement === false) {
+      // Set tabIndex="0" so focus can go into the document when
+      // using tab key when scrolling is enabled
+      document.body.tabIndex = 0;
+    }
```

Comment:
I was doing integration testing and I noticed that sometimes on slow connection, the focus isn't set perhaps due to `tabIndex=0` isn't applied. I think I track it down to the call above being slow to respond or something like that. I tried moving this code into the `.then()` and that seems to be working now.

I'll update and move it in there.

Reply:
I updated the logic here